src/nvim/main.c

```diff
@@ -2070,3 +2073,35 @@ static void check_swap_exists_action(void)
     getout(1);
   handle_swap_exists(NULL);
 }
+
+
+/// Callback used by `check_vimruntime`
+static void check_vimruntime_callback(char_u *fname, void *cookie)
+{
+  STRNCPY(NameBuff, fname, MAXPATHL);
+}
+
+/// Check if $VIMRUNTIME points to a valid Neovim runtime path, look for
+/// $VIMRUNTIME/autoload/provider/clipboard.vim, which is a Neovim-specific
+/// file. Print a warning if the file is not found or not readable.
+static void check_vimruntime(void)
+{
+  int result = do_in_runtimepath((char_u*) "autoload/provider/clipboard.vim",
+                                 false, check_vimruntime_callback, NULL);
+
+  if (result == FAIL) {
+    const char *runtime = os_getenv("VIMRUNTIME");
+    if (runtime) {
+      vim_snprintf((char*) IObuff, IOSIZE,
+                   _("W51: Invalid VIMRUNTIME: \"%s\""), runtime);
```

Comment:
This error message is invalid. The fact that VIMRUNTIME has some non-empty value does *not* mean that it is present in runtimepath.